lib/starknet-jvm.md

```diff
@@ -236,21 +238,72 @@ Data classes representing StarkNet transactions.
 
 # Package com.swmansion.starknet.provider
 
-Provider interface used for interacting with StarkNet.
+Provider interface and its implementations.
+
+```java
+// Create a provider instance
+Provider provider = ...
+
+// Get a storage value request
+Request<Felt> request = provider.getStorageAt(address, key);
+// Send a request
+Felt response = request.send();
+
+// For most methods block hash, number or tag can be specified
+Request<Felt> request = provider.getStorageAt(address, key, Felt.fromHex("0x123..."));
+Request<Felt> request = provider.getStorageAt(address, key, 1234);
+Request<Felt> request = provider.getStorageAt(address, key, BlockTag.LATEST);
+```
 
 # Package com.swmansion.starknet.provider.exceptions
 
 Exceptions thrown by the StarkNet providers.
 
+`Request.send()` throws `RequestFailedException` unchecked exception.
+It can optionally be handled.
+
+```java
+Request<Felt> request = ...
+// Send a request
+try {
+    Felt response = request.send();
+} catch (RequestFailedException e) {
+    // Handle an exception
+}
+```
+
+In the case of `Request.sendAsync()`, an exception would have to be handled in the returned `CompletableFuture`.
+
 # Package com.swmansion.starknet.provider.gateway
 
 Provider utilising StarkNet gateway and feeder gateway for communication with the network.
 
+```java
+// Create a provider using a factory
```

Comment:
This is not really a factory. It might be called factory method. I like to call methods like this named constructors.

Reply:
Let's just call it static method